feat: more readme (#16)
BitcoinL2-Labs/sPoX: README.md

# sPoX

`sPoX` is a binary that monitors the Bitcoin blockchain for sBTC deposits made to a set of addresses, and when found
informs Emily about them so the sBTC signers can then process them.

> Its primary usage is to enable automatic bridging of Stacks POX payments to sBTC.

Reviewer: We should add a medium level overview of how it works. I don't mind adding something here.

## Building

To build `sPoX`, run:
```bash
cargo build --bin spox --release --locked
```

The binary will be built in `target/release/spox`.

## Configuration

You can specify which deposits to look for and the endpoints to use in a toml file.
See `src/config/default.toml` for a config starting point. Note that the Stacks configuration is required only to run
the `get-signers-xonly-key` command.

### Get signers xonly public key

> When configuring a deposit, you must specify for what signers public key it is for in `signers_xonly`. This key changes

Reviewer: ```suggestion
When configuring a deposit, you must specify the sBTC signers' public key using the `signers_xonly` field in the config. This key changes
```

over time after sBTC key rotations. To fetch the current key, fill the `stacks` stanza with the Stacks endpoint and 

> deployer address (for Stacks mainnet, see https://github.com/stacks-sbtc/sbtc/blob/main/docker/mainnet/sbtc-signer/signer-config.toml.in#L109).

Reviewer: I would pin this to a particular tag.

Then you can run:
```bash
./spox -c <config file> get-signers-xonly-key
```

> to get the latest key from the sBTC registry smart contract.

Reviewer: I always find it helpful to mention whether the path here can be relative, absolute, or both.

### Get a deposit address

Once you have configured a deposit, you can run:
```bash
./spox -c <config file> get-deposit-address
```

> To get the deposit address, for each configured deposit.

Reviewer: ```suggestion
to get the bitcoin address, for each configured deposit.
```

## Run `sPoX`

Once the configuration is completed, you can run `sPoX`:
```bash
./spox -c <config file>
```
The binary will monitor the Bitcoin blockchain for payments made to the monitored addresses, and when a new payment is

> confirmed, it will notify Emily about it so that the sBTC can process it.

Reviewer: ```suggestion
confirmed, it will notify Emily about it so that the sBTC signers can process it.
```
